src/foam/nanos/http/AuthWebAgent.java

```diff
@@ -169,7 +171,15 @@ public Session authenticate(X x) {
           .put(HttpServletRequest.class,  req)
           .put(HttpServletResponse.class, resp), email, password);
 
-        if ( user != null ) return session;
+        if ( user != null ) {
+          if ( ! SafetyUtil.isEmpty(actAs) ) {
+            AgentAuthService agentService = (AgentAuthService) x.get("agentAuth");
+            DAO localUserDAO = (DAO) x.get("localUserDAO");
+            User entity = (User) localUserDAO.find(Long.parseLong(actAs));
```

Comment:
Could this throw an exception if I put `&actAs=string` in the query instead of a valid long? Should we catch an handle that by throwing a descriptive error message?

Reply:
YES!

Reply:
Have addressed this, but not comprehensively from a DIG point of view. Have created a different task for that.

Reply:
So Joel and I talked about this, I fixed this locally, but it still doesn't write the appropriate message to an api consumer. Going to address that in a different pr.